src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -227,6 +227,8 @@ private static ReportDiagnostic MapSeverityToReport(DiagnosticSeverity severity)
                     return ReportDiagnostic.Warn;
                 case DiagnosticSeverity.Error:
                     return ReportDiagnostic.Error;
+                case DiagnosticSeverity.Suggestion:
```

Comment:
Suggestion: Diagnostics that the user wants to see in the editor but does not want to break their build. For example, code style diagnostics (e.g. use var or not) would be suggestions by default. These can be elevated to warnings or errors if the user chooses. However, they are a fully supported severity level.

Info: Provides informational data, but has an existing meaning and is therefore not suitable.

Hidden: A diagnostic that is really a bit of a design wart. Essentially, it's something that is intended to represent a span that does not appear in the editor, but the user can hang a code fix on it.

That's how I see them.